Python Updates 2022-03-30

mweinelt opened the pull request:
Likely the last python updates series before the upcoming zero hydra failures period.

https://hydra.nixos.org/job/nixpkgs/python-updates/tested#tabs-constituents

The previous cycle was #162538 and is currently still in staging-next due to delays caused by the latest OpenSSL update and problems around x86_64-darwin on hydra.

Latest eval report:
https://malob.github.io/nix-review-tools-reports/nixpkgs:python-updates/nixpkgs_python-updates_1755257.html

dotlambda committed:
python3Packages.rapidfuzz: 1.9.1 -> 2.0.8

https://github.com/maxbachmann/RapidFuzz/blob/v2.0.8/CHANGELOG.md

SuperSandro2000 committed:
python39Packages.hypothesmith: remove linting programs

SuperSandro2000 committed:
python39Packages.ipython: remove optional black depedency, update homepage

Black should be updatable without a mass rebuild

SuperSandro2000 committed:
python39Packages.typer: remove linting programs

SuperSandro2000 committed:
python39Packages.bcrypt: remove unused dependencies

SuperSandro2000 committed:
python39Packages.flask: fix build & add SuperSandro2000 as maintainer

fabaff committed:
python3Packages.bcrypt: add pythonImportsCheck

Artturin committed:
python39Packages.pytest-xdist: add pexpect to run more tests

Artturin committed:
python39Packages.pytest-xdist: add hook to add flags

many of our packages have pytest-xdist in their inputs however
oftentimes the required flags aren't added to pytest flags
since we dont use tox and other frameworks

some examples of improvements
(my pc has ryzen 9 5900x)
hypothesis 11m:15s on hydra -> <1m on my pc
hypothesmith 8m on my pc -> <1m on my pc

the whole graph from pytest-xdist (tested by adding postPatch with true to xdist)
time nix build ".#python3Packages.hypothesmith"
19m:57s -> 9m:13s

mweinelt committed:
python3Packages.kivy-garden: 0.1.4 -> 0.1.5

mweinelt committed:
python3Packages.jupyter-packaging: 0.11.1 -> 0.12.0

mweinelt committed:
python3Packages.jupyterlab_server: 2.10.3 -> 2.12.0

mweinelt committed:
python3Packages.jupyterlab-git: 0.34.2 -> 0.36.0

mweinelt committed:
python3Packages.Mako: 1.1.6 -> 1.2.0

mweinelt committed:
python3Packages.marshmallow-sqlalchemy: 0.27.0 -> 0.28.0

mweinelt committed:
python3Packages.mccabe: 0.6.1 -> 0.7.0

mweinelt committed:
python3Packages.mailchecker: 4.1.13 -> 4.1.14

mweinelt committed:
python3Packages.limits: 2.3.3 -> 2.4.0

mweinelt committed:
python3Packages.moto: 3.0.5 -> 3.1.3

mweinelt committed:
python3Packages.metakernel: 0.28.2 -> 0.29.0

mweinelt committed:
python3Packages.nbformat: 5.1.3 -> 5.2.0

mweinelt committed:
python3Packages.netmiko: 3.4.0 -> 4.0.0

mweinelt committed:
python3Packages.notebook: 6.4.8 -> 6.4.10

mweinelt committed:
python3Packages.numpydoc: 1.2 -> 1.2.1

mweinelt committed:
python3Packages.nbclient: 0.5.11 -> 0.5.13

mweinelt committed:
python3Packages.nbconvert: 6.4.2 -> 6.4.5

mweinelt committed:
python3Packages.networkx: 2.7 -> 2.7.1

mweinelt committed:
python3Packages.mypy-boto3-s3: 1.21.7 -> 1.21.27.post1

mweinelt committed:
python3Packages.nunavut: 1.7.3 -> 1.7.5

mweinelt committed:
python3Packages.phonemizer: 3.0.1 -> 3.1.1

mweinelt committed:
python3Packages.phik: 0.12.0 -> 0.12.2

mweinelt committed:
python3Packages.ovh: 0.5.0 -> 1.0.0

mweinelt committed:
python3Packages.pdm-pep517: 0.11.2 -> 0.12.1

mweinelt committed:
python3Packages.prettytable: 3.1.1 -> 3.2.0

mweinelt committed:
python3Packages.pymc3: 3.11.4 -> 3.11.5

mweinelt committed:
python3Packages.pygit2: 1.9.0 -> 1.9.1

mweinelt committed:
python3Packages.pycountry: 22.1.10 -> 22.3.5

mweinelt committed:
python3Packages.pyfakefs: 4.5.5 -> 4.5.6

mweinelt committed:
python3Packages.pybids: 0.14.0 -> 0.15.0

mweinelt committed:
python3Packages.pykerberos: 1.2.3.dev0 -> 1.2.4

mweinelt committed:
python3Packages.py4j: 0.10.9.3 -> 0.10.9.5

mweinelt committed:
python3Packages.pytest-metadata: 1.11.0 -> 2.0.1

mweinelt committed:
python3Packages.pyro-ppl: 1.8.0 -> 1.8.1

mweinelt committed:
python3Packages.pytest-testmon: 1.3.0 -> 1.3.1

mweinelt committed:
python3Packages.python-dotenv: 0.19.2 -> 0.20.0

mweinelt committed:
python3Packages.pytest-check: 1.0.4 -> 1.0.5

mweinelt committed:
python3Packages.python-logstash: 0.4.6 -> 0.4.8

mweinelt committed:
python3Packages.pyramid_jinja2: 2.8 -> 2.10

mweinelt committed:
python3Packages.pytools: 2022.1 -> 2022.1.2

mweinelt committed:
python3Packages.redis: 4.1.4 -> 4.2.0

mweinelt committed:
python3Packages.PyWavelets: 1.2.0 -> 1.3.0

mweinelt committed:
python3Packages.qtconsole: 5.2.2 -> 5.3.0

mweinelt committed:
python3Packages.pytz: 2021.3 -> 2022.1

mweinelt committed:
python3Packages.readthedocs-sphinx-ext: 2.1.4 -> 2.1.5

mweinelt committed:
python3Packages.reportlab: 3.6.8 -> 3.6.9

mweinelt committed:
python3Packages.spyder: 5.2.2 -> 5.3.0

mweinelt committed:
python3Packages.spyder-kernels: 2.2.1 -> 2.3.0

mweinelt committed:
python3Packages.Shapely: 1.8.0 -> 1.8.1.post1

mweinelt committed:
python3Packages.sphinx-jinja: 1.4.0 -> 2.0.1

mweinelt committed:
python3Packages.SPARQLWrapper: 1.8.5 -> 2.0.0

mweinelt committed:
python3Packages.sagemaker: 2.77.1 -> 2.82.0

mweinelt committed:
python3Packages.spacy: 3.2.3 -> 3.2.4

mweinelt committed:
python3Packages.spacy-transformers: 1.1.4 -> 1.1.5

mweinelt committed:
python3Packages.torchinfo: 1.6.3 -> 1.6.5

mweinelt committed:
python3Packages.tifffile: 2022.2.9 -> 2022.3.25

mweinelt committed:
python3Packages.tomlkit: 0.10.0 -> 0.10.1

mweinelt committed:
python3Packages.tpm2-pytss: 1.0.0 -> 1.1.0

mweinelt committed:
python3Packages.svg.path: 4.1 -> 5.1

mweinelt committed:
python3Packages.terminado: 0.13.1 -> 0.13.3

mweinelt committed:
python3Packages.sympy: 1.9 -> 1.10.1

mweinelt committed:
python3Packages.urllib3: 1.26.8 -> 1.26.9

mweinelt committed:
python3Packages.typer: 0.4.0 -> 0.4.1

mweinelt committed:
python3Packages.virtualenv: 20.13.2 -> 20.14.0

mweinelt committed:
python3Packages.ufo2ft: 2.25.3 -> 2.26.0

mweinelt committed:
python3Packages.tzdata: 2021.5 -> 2022.1

mweinelt committed:
python3Packages.uamqp: 1.5.1 -> 1.5.3

mweinelt committed:
python3Packages.tqdm: 4.63.0 -> 4.63.1

mweinelt committed:
python3Packages.waitress: 2.0.0 -> 2.1.1

mweinelt committed:
python3Packages.zope.testing: 4.9 -> 4.10

mweinelt committed:
python3Packages.widgetsnbextension: 3.5.2 -> 3.6.0

mweinelt committed:
python3Packages.werkzeug: 2.0.3 -> 2.1.0

mweinelt committed:
python3Packages.zarr: 2.11.0 -> 2.11.1

mweinelt committed:
python3Packages.ZODB: 5.6.0 -> 5.7.0

mweinelt committed:
python3Packages.websocket-client: 1.3.1 -> 1.3.2

mweinelt committed:
python3Packages.watchgod: 0.7 -> 0.8.1

mweinelt committed:
python3Packages.watchdog: 2.1.6 -> 2.1.7

mweinelt committed:
python3Packages.sphinx: 4.4.0 -> 4.5.0

https://www.sphinx-doc.org/en/master/changes.html\#release-4-5-0-released-mar-28-2022

mweinelt committed:
python3Packages.flake8: relax mccabe constraint

mweinelt committed:
python3Packages.mccabe: disable broken tests

mweinelt committed:
python3Packages.ephemeral-port-reserve: init at 1.1.4

mweinelt committed:
python3Packages.werkzeug: fix tests

mweinelt committed:
python3Packages.watchgod: add missing anyio dependency

mweinelt committed:
python3Packages.redis: add missing deps, set up extra-requires

mweinelt committed:
python3Packages.pylint: 2.12.2 -> 2.13.3

mweinelt committed:
python3Packages.astroid: 2.9.3 -> 2.11.2

mweinelt committed:
python3Packages.asttokens: move astroid to check inputs, disable astroid tests

mweinelt committed:
python3Packages.pylint: add missing deps, adjust pytest-xdist param

SuperSandro2000 commented on pkgs/development/python-modules/ansible-lint/default.nix:
```diff
@@ -18,13 +18,13 @@
 buildPythonPackage rec {
   pname = "ansible-lint";
-  version = "5.4.0";
+  version = "6.0.2";
```
That one was a bit more involved and was already done in https://github.com/NixOS/nixpkgs/pull/166283

SuperSandro2000 commented on pkgs/development/python-modules/elasticsearch/default.nix:
```diff
@@ -12,11 +12,11 @@ buildPythonPackage (rec {
   # there's a clear path forward. See
   # https://github.com/elastic/elasticsearch-py/issues/1639 for more
   # info.
-  version = "7.16.3";
```
If we follow the comment above we should downgrade to 7.13.X.

SuperSandro2000 commented on pkgs/development/python-modules/sagemaker/default.nix:
```diff
@@ -17,14 +17,14 @@
 buildPythonPackage rec {
   pname = "sagemaker";
-  version = "2.77.1";
+  version = "2.82.0";
```
I think this is sage related and we should skip updating it.

SuperSandro2000 committed:
python310Packages.sparqlwrapper: normalize attr

mweinelt committed:
python3Packages.ipykernel: 6.10.0 -> 6.11.0

mweinelt committed:
python3Packages.python-vagrant: 0.5.15 -> 1.0.0

mweinelt committed:
python3Packages.redis: 4.2.0 -> 4.2.1

mweinelt committed:
python3Packages.parsimonious: 0.8.1 -> 0.9.0

mweinelt committed:
python3Packages.twine: 3.8.0 -> 4.0.0

mweinelt committed:
python3Packages.pytest: prune deps, refactor

mweinelt committed:
python3Packages.purl: propagate six, add import check

mweinelt committed:
python3Packages.paho-mqtt: require six for tests

SuperSandro2000 committed:
python39Packages.rich: 11.2.0 -> 12.0.1

mweinelt committed:
python3Packages.pymemcache: propagate six

mweinelt committed:
python3Packages.reportlab: patch path to freetype headers

mweinelt committed:
python3Packages.sqlalchemy: 1.4.32 -> 1.4.33

https://github.com/sqlalchemy/sqlalchemy/releases/tag/rel_1_4_33

mweinelt committed:
python3Packages.Mako: add babel extra, disable lingua tests

mweinelt committed:
python3Packages.service-identity: propagate six

mweinelt committed:
python3Packages.astroid: disable test_mro_typing_extensions on python310

mweinelt committed:
python3Packages.beautifultable: add wcwidth dep, fix homepage

mweinelt committed:
python3Packages.claripy: add six to check deps

mweinelt committed:
python3Packages.fasteners: add more-itertools to check deps

mweinelt committed:
python3Packages.ush: add six to check deps

mweinelt committed:
python3Packages.opentracing: add six to check deps, migrate to pytestCheckHook

mweinelt committed:
lookatme: fix tests

mweinelt committed:
python3Packages.dictpath: add six to check deps

mweinelt committed:
python3Packages.nocasedict: propagate six, migrate to pytestCheckHook

mweinelt committed:
python3Packages.parsimonious: fix build, use pytestCheckHook

mweinelt committed:
python3Packages.validators: propagate six

mweinelt committed:
python3Packages.variants: add six to check deps

mweinelt committed:
python3Packages.python-registry: add six to check deps

mweinelt committed:
python3Packages.aurorapy: add six to check deps

mweinelt committed:
python310Packages.validictory: disable

mweinelt committed:
python3Packages.flake8-blind-except: propagate pycodestyle

mweinelt committed:
python3Packages.sqlalchemy: 1.4.33 -> 1.4.34

mweinelt committed:
python3Packages.wasm: disable

mweinelt committed:
python310Packages.py-radix: disable

mweinelt committed:
python3Packages.reedsolo: use git tag

mweinelt committed:
python3Packages.reedsolo: python 3.10 compat

mweinelt committed:
python3Packages.flask-paranoid: disable failing tests

mweinelt committed:
Revert "python3Packages.regex: 2022.3.2 -> 2022.3.15"

This reverts commit 1be45287bdedb40f3d070733dc5975f64341c9c9.

Completely breaks dateparser.

mweinelt committed:
python3Packages.dateparser: 1.1.0 -> 1.1.1

mweinelt committed:
python3Packages.pympler: disable failing test

mweinelt committed:
python3Packages.flask-security-too: split off extra-requires

mweinelt committed:
python3Packages.flask-security-too: disable tests incompatible with flask>=2.1.0

mweinelt committed:
python3Packages.numpydoc: relax jinja2 constraint, disable failing tests

mweinelt committed:
python3Packages.fn: patch for python3.10 compat

mweinelt committed:
python3Packages.importlib-resources: use pyproject format

mweinelt committed:
python3Packages.nunavut: propagate importlib-resources

mweinelt committed:
python3Packages.ovh: propagate requests

mweinelt committed:
python3Packaegs.pyspark: update py4j pin relaxation

mweinelt committed:
python3Packages.django-modelcluster: rename, fix build, enable tests

mweinelt committed:
python3Packaegs.sphinx_rtd_theme: update docutils pin relaxation

mweinelt committed:
python3Packages.clize: relax docutils constraint, set up extras-require

mweinelt committed:
python3Packages.dugong: disable tests on python310

mweinelt committed:
python3Packages.pytest-xdist: fix hook being applied multiple times

mweinelt committed:
python3Packages.moto: update disabled tests, drop xdist

Dropping pytest-xdist due to reproducible hangs at the end of the test
run.

mweinelt committed:
python3Packages.flask-paranoid: 0.2 -> 0.3.0

mweinelt committed:
python3Packages.pandas: 1.4.1 -> 1.4.2

mweinelt committed:
python3Packages.funcparserlib: add six to check deps

mweinelt committed:
python3Packages.httplib2: drop xdist, prone to race conditions

risicle commented:
Rebased & dropped the superseded commits

mweinelt committed:
python3Packages.pylint: 2.13.3 -> 2.13.4

risicle commented:
Flask-RESTful appears broken due to https://github.com/flask-restful/flask-restful/issues/936. There's a fix, but not merged yet.

mweinelt committed:
python3Packages.flask-restful: apply patch for werkzeug 2.1.0 compat

Based on https://github.com/python-restx/flask-restx/pull/423.

mweinelt commented:
> Flask-RESTful appears broken due to [flask-restful/flask-restful#936](https://github.com/flask-restful/flask-restful/issues/936). There's a fix, but not merged yet.

Adapted a change from flask-restx in [`ee355b0` (#166489)](https://github.com/NixOS/nixpkgs/pull/166489/commits/ee355b0e0ebfd40e352fdd1ab79700f7e2ecc3b8).

mweinelt committed:
python3Packages.httpbin: apply patch for werkzeug 2.1.0 compatibility

mweinelt committed:
pytest-httpbin: drop assertion that doesn't hold anymore

In Flask 2.1.0 Location headers were made relative and alot of packages
still assume absolute values. That breaks the assertion, so we drop it.

risicle committed:
python3Packages.asgiref: fix tests on darwin

our macos sdk version doesn't have SOCK_NONBLOCK

risicle committed:
python3Packages.ephemeral-port-reserve: skip test_fqdn on darwin

mweinelt commented:
@ofborg eval

mweinelt commented:
Can we ask upstream about making black tests optional instead of carrying that patch?

risicle commented:
/me considers writing a no-op dummy `black` package.

risicle commented:
`mitmproxy` appears broken, will have to investigate tomorrow.

mweinelt committed:
python3Packages.ipykernel: 6.11.0 -> 6.12.1

mweinelt committed:
python3Packages.botocore: 1.24.30 -> 1.24.33

mweinelt committed:
awscli: 1.22.67 -> 1.22.88

mweinelt committed:
python3Packages.azure-core: 1.23.0 -> 1.23.1

mweinelt committed:
python3Packages.azure-mgmt-core: propagate typing-extensions

mweinelt committed:
python3Packages.nbconvert: propagate beautifulsoup4

mweinelt committed:
python3Packages.pdm-pep517: 0.12.1 -> 0.12.3

SuperSandro2000 committed:
python39Packages.cffi: move prePatch to postPatch to not break patches

FRidh committed:
python3.pkgs.bcrypt: fix build

also use pyproject format as we're heading that direction.

SuperSandro2000 committed:
python39Packages.yanc: only disable tests on python >=3.5

dtzWill committed:
python3Packages.pycurl: re-enable tests that no longer fail

Artturin committed:
python39Packages.pytest-xdist: run xdist hook before pytestCheckPhase

risicle committed:
python3Packages.shapely: update patch to apply to 1.8.1.post1

mweinelt committed:
python3Packages.httplib2: 0.20.3 -> 0.20.4

mweinelt committed:
python3Packages.sqlalchemy: 1.4.34 -> 1.4.35

mweinelt committed:
python3Packages.pandas: use upstreams fast test settings

dotlambda committed:
python3Packages.img2pdf: 0.4.3 -> 0.4.4

https://gitlab.mister-muffin.de/josch/img2pdf/src/tag/0.4.4/CHANGES.rst

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/105.html

grpc fix over in #167129

mweinelt committed:
python3Packages.junos-eznc: fix build and tests

mweinelt commented:
https://shells.darmstadt.ccc.de/~hexa/107.html

risicle committed:
python3Packages.mitmproxy: add dontUsePytestXdist flag

a couple of tests don't seem to like this, producing confusing
failures

SuperSandro2000 commented:
> [shells.darmstadt.ccc.de/~hexa/107.html](https://shells.darmstadt.ccc.de/~hexa/107.html)

Would be nice if we could link this in the first comment otherwise you need to expand multiple times the github thread to be able to see it.

dotlambda committed:
python3Packages.pillow: 9.0.1 -> 9.1.0

https://pillow.readthedocs.io/en/stable/releasenotes/9.1.0.html

dotlambda committed:
python3Packages.pikepdf: disable failing test

mweinelt committed:
python3Packages.google-cloud-testutils: propagate packaging

jonringer committed:
python3Packges.websocket: add missing six dependency

```
       > Processing ./websocket_client-0.56.0-py2.py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement six (from websocket-client) (from versions: none)
       > ERROR: No matching distribution found for six
       >
```

risicle committed:
python3Packages.rich: 12.0.1 -> 12.2.0

mweinelt committed:
python3Packages.jax: test with limited parallelism

The tests are prone to getting stuck with high parallelism.

risicle committed:
python3Packages.fastparquet: 0.7.1 -> 0.8.1

SuperSandro2000 committed:
python39Packages.ncclient: 0.6.12 -> 0.6.13

SuperSandro2000 committed:
python310Packages.click: 8.1.1 -> 8.1.2

mweinelt committed:
python3Packages.jax: disable test_custom_linear_solve_aux

```
______________ CustomLinearSolveTest.test_custom_linear_solve_aux ______________
[gw3] linux -- Python 3.9.11 /nix/store/k1physzalj5vffsvl7ag6h6b6vaqip5x-python3-3.9.11/bin/python3.9

self = <custom_linear_solve_test.CustomLinearSolveTest testMethod=test_custom_linear_solve_aux>

    @jtu.skip_on_flag("jax_skip_slow_tests", True)
    def test_custom_linear_solve_aux(self):
      def explicit_jacobian_solve_aux(matvec, b):
        x = lax.stop_gradient(jnp.linalg.solve(jax.jacobian(matvec)(b), b))
        return x, array_aux

      def matrix_free_solve_aux(matvec, b):
        return lax.custom_linear_solve(
          matvec, b, explicit_jacobian_solve_aux, explicit_jacobian_solve_aux,
          symmetric=True, has_aux=True)

      def linear_solve_aux(a, b):
        return matrix_free_solve_aux(partial(high_precision_dot, a), b)

      # array aux values, to be able to use jtu.check_grads
      array_aux = {"converged": np.array(1.), "nfev": np.array(12345.)}
      rng = self.rng()
      a = rng.randn(3, 3)
      a = a + a.T
      b = rng.randn(3)

      expected = jnp.linalg.solve(a, b)
      actual_nojit, nojit_aux = linear_solve_aux(a, b)
      actual_jit, jit_aux = jax.jit(linear_solve_aux)(a, b)

      self.assertAllClose(expected, actual_nojit)
      self.assertAllClose(expected, actual_jit)
      # scalar dict equality check
      self.assertDictEqual(nojit_aux, array_aux)
      self.assertDictEqual(jit_aux, array_aux)

      # jvp / vjp test
>     jtu.check_grads(linear_solve_aux, (a, b), order=2, rtol=4e-3)

tests/custom_linear_solve_test.py:157:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
jax/_src/test_util.py:372: in check_grads
    _check_grads(f, args, order)
jax/_src/test_util.py:361: in _check_grads
    _check_grads(partial(api.jvp, f), (args, args), order - 1, fwd_msg)
jax/_src/test_util.py:365: in _check_grads
    _check_vjp(f, partial(api.vjp, f), args, err_msg=rev_msg)
jax/_src/test_util.py:325: in check_vjp
    check_close(ip, ip_expected, atol=atol, rtol=rtol,
jax/_src/test_util.py:227: in check_close
    tree_all(tree_multimap(assert_close, xs, ys))
jax/_src/tree_util.py:180: in tree_map
    return treedef.unflatten(f(*xs) for xs in zip(*all_leaves))
jax/_src/tree_util.py:180: in <genexpr>
    return treedef.unflatten(f(*xs) for xs in zip(*all_leaves))
jax/_src/test_util.py:217: in _assert_numpy_close
    _assert_numpy_allclose(a, b, atol=atol * a.size, rtol=rtol * b.size,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

a = array(1.89683694), b = array(1.88698006), atol = 0.002, rtol = 0.004
err_msg = 'VJP of JVP cotangent projection'

    def _assert_numpy_allclose(a, b, atol=None, rtol=None, err_msg=''):
      if a.dtype == b.dtype == _dtypes.float0:
        np.testing.assert_array_equal(a, b, err_msg=err_msg)
        return
      a = a.astype(np.float32) if a.dtype == _dtypes.bfloat16 else a
      b = b.astype(np.float32) if b.dtype == _dtypes.bfloat16 else b
      kw = {}
      if atol: kw["atol"] = atol
      if rtol: kw["rtol"] = rtol
      with np.errstate(invalid='ignore'):
        # TODO(phawkins): surprisingly, assert_allclose sometimes reports invalid
        # value errors. It should not do that.
>       np.testing.assert_allclose(a, b, **kw, err_msg=err_msg)
E       AssertionError:
E       Not equal to tolerance rtol=0.004, atol=0.002
E       VJP of JVP cotangent projection
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 0.00985688
E       Max relative difference: 0.00522363
E        x: array(1.896837)
E        y: array(1.88698)

jax/_src/test_util.py:187: AssertionError
```

dotlambda committed:
python3Packages.pytest-asyncio: 0.18.1 -> 0.18.3

https://github.com/pytest-dev/pytest-asyncio/blob/v0.18.3/CHANGELOG.rst

dotlambda committed:
python3Packages.async_generator: does not depend on pytest-asyncio

mweinelt committed:
treewide: remove redundant numprocesses pytest-xdist parameter

Including the pytest-xdist package automatically configures numprocesses
to $NIX_BUILD_CORES, so specifying these on a package level has become
redundant.

mweinelt committed:
borgbackup: disable racy test

https://github.com/borgbackup/borg/issues/6573

mweinelt committed:
python3Packages.twisted: 22.2.0 -> 22.4.0

mweinelt committed:
python3Packages.path: rename from path.py; 12.0.1 -> 16.4.0

The project had been renamed after 12.5.0 back in 2020, which is why we
didn't receive further updates.

Created an alias and renamed all existing references.

mweinelt committed:
python3Packges.pep257: remove together with pytest-pep257

The code apparently was migrated into pycodestyle.

https://github.com/PyCQA/pydocstyle/issues/580

mweinelt committed:
python3Packages.mkdocs: 1.2.3 -> 1.3.0

https://github.com/mkdocs/mkdocs/releases/tag/1.3.0

mweinelt committed:
python3Packages.mkdocs-material: 8.2.7 -> 8.2.9

mweinelt committed:
python3Packages.fakeredis: relax redis constraint

kirillrdy committed:
awsebcli: fixup, downgrade jmespath

mweinelt commented:
Git currently fails to build on staging so evals are failing early.

mweinelt committed:
python3Packages.pytest-shutil: fix build

Propagates six and replaces path.py dependency with path, the new name.

mweinelt committed:
python3.pkgs.pylint: 2.13.4 -> 2.13.5

https://github.com/PyCQA/pylint/releases/tag/v2.13.5

mweinelt committed:
grab-site: update sqlalchemy hash

mweinelt commented:
https://malob.github.io/nix-review-tools-reports/nixpkgs:python-updates/nixpkgs_python-updates_1755257.html

dotlambda commented on pkgs/development/python-modules/pylint/default.nix:
```diff
@@ -74,6 +75,9 @@ buildPythonPackage rec {
     "test_py3k_jobs_option"
+  ] ++ lib.optionals (pythonAtLeast "3.10") [
+    # https://github.com/PyCQA/pylint/issues/6296
+    "test_functional"
```
Why don't we add `typing-extensions` to `checkInputs`?

mweinelt replied:
Yep, I have that already fixed up locally, will be part of the next push.

mweinelt commented:
Removed pyenchant from pylints check inputs because it introduces a dependency on nuspell/pandoc which is a bit heavy for my taste.

mweinelt committed:
python3Packages.falcon: fix build; cythonize!

mweinelt committed:
mailman: fix build

Required an update to the sqlalchemy override due to new hash variable
usage. Also disables tests for sqlalchemy and alembic because neither
can find any, so they error out.

mweinelt committed:
buku: update sqlalchemy hash

mweinelt committed:
home-assistant: reduce component tests processes

Component tests are often of lesser quality, which may lead to spurious
errors when run with high parallelization. Reducing to 4 processes
aligns the setup with what GitHub actions provides, which is what
upstream uses to test.

mweinelt committed:
octoprint: fix eval

Relying on attributes like these to always be available is prone to an
eval error when the overridden package drops them.

mweinelt committed:
home-assistant-cli: downgrade click to 8.0.4

https://github.com/home-assistant-ecosystem/home-assistant-cli/issues/382

mweinelt committed:
python3Packages.azure-core: propagate typing-extensions

mweinelt committed:
python3Packages.starlette: 0.18.0 -> 0.19.0

mweinelt committed:
python3Packages.python3-saml: relax lxml constraint

mweinelt committed:
python3Packages.approvaltests: 4.0.0 -> 5.0.0

mweinelt committed:
python3Packages.flask-socketio: 5.0.1 -> 5.1.1

mweinelt committed:
python3Packages.glean-parser: relax markupsafe constraint

mweinelt committed:
python3Packages.atom: fix versioning and tests

dotlambda committed:
python3Packages.rapidfuzz-capi: init at 1.0.5

dotlambda committed:
python3Packages.jarowinkler: init at 1.0.2

mweinelt committed:
fava: 1.19 -> 1.21; fix Werkzeug compat

FRidh committed:
python3Packages.jaxlib: set platforms

mweinelt committed:
python3Packages.kivy-garden: fix source

mweinelt committed:
python3Packages.ansible-runner: build requires pbr

mweinelt committed:
python3Packages.datasette: relax click & jinja2 constraints

Move substituion into postPatch, the canonical place for this kind of
operation.

mweinelt committed:
python3Packages.matchpy: 0.5.1 -> 0.5.5

mweinelt committed:
python3Packages.scikit-build: fix build

Drops code that references non-existant distutils object and then:

- Migrates to pyproject,
- uses pytestCheckHook,
- trims checkInputs.

mweinelt committed:
python3Packages.sanic: disable test_noisy_exceptions

Sometimes the noisy exceptions output is missing, which makes the test flaky.

mweinelt committed:
python3Packages.chalice: relax jmespath constraint

https://github.com/aws/chalice/commit/48511562f189d33b43438ca608b10fc376508f09

mweinelt committed:
python3Packages.trytond: fix werkzeug2.1 compat in test suite

mweinelt committed:
python3Packages.pytorch-bin: don't build on hydra

It won't complete anyhow, as it exceeds the recently increased maximum
output size by another 200 MB.

https://hydra.nixos.org/build/173626214

mweinelt committed:
python3Packages.pyslurm: 19-05-0 -> 21.08.4

Unbreaks the package.

mweinelt committed:
python3Packages.pyrad: revamp

- Use pyproject build
- Patch poetry-core usage
- Use unittest over nosetest
- Fix meta.homepage

mweinelt committed:
python3Packages.SQLAlchemy-ImageAttach: drop

Unmaintained since 2017 and incompatible with SQLAlchemy 1.4

mweinelt committed:
python3Packages.flask-restx: patch for werkzeug 2.1 compat

mweinelt committed:
python3Packages.gipc: remove broken flag

mweinelt committed:
python3Packages.prompt_toolkit: propagate six

mweinelt committed:
python3Packages.vdirsyncer: relax click-log constraint

mweinelt committed:
python3Packages.json-schema-for-humans: relax pytz constraint

mweinelt committed:
litecli: 1.6.0 -> 1.8.0

mweinelt committed:
python3Packages.oci: 2.60.0 -> 2.63.0

mweinelt committed:
oci-cli: 3.6.0 -> 3.7.2

Pins jmespath to 0.10.0.

mweinelt committed:
python3Packages.myfitnesspal: propagate typing-extensions

mweinelt committed:
python3Packages.installer: 0.3.0 -> 0.5.1

mweinelt committed:
pdm: 1.13.3 -> 1.14.0

mweinelt committed:
pgadmin4: relax pytz constraint

mweinelt committed:
python3Packages.grip: 4.5.2 -> 4.6.1

SuperSandro2000 committed:
python310Packages.cython_3: init at 3.0.0a10

SuperSandro2000 committed:
python310Packages.levenshtein: fix build

SuperSandro2000 committed:
python310Packages.scikit-build: 0.13.1 -> 0.14.1

mweinelt committed:
puddletag: relax lxml version
